### DISCUSSION TITLE
Intro documentation about async

### DISCUSSION THREAD

**Discussion by kassane** [alibaba/async_simple#346] [General]
I have recently seen, in addition to this project (based on `asio` standalone), and a new proposal for `boost::cobalt` (based on `boost::asio`) focused on single-threaded `io_context` in main function.

Home: https://boostorg.github.io/cobalt/
Src: https://github.com/boostorg/cobalt

Since they have this superficial similarity. 

### Is this project intended to detail the internal functioning of asio or explain a more detailed about async on C++?

Unfortunately, many are unhappy with the official library documentation (boost or standalone).

**Comment by ChuanqiXu9** [comment]
Hi Matheus,

> in addition to this project (based on asio standalone)
> Is this project intended to detail the internal functioning of asio

This is incorrect. This project is not based on asio. We just use asio in our demo examples.

> or explain a more detailed about async on C++?

One of the key ideas of the project is to left the implementation of executors to the users so that the users can control what happens actually. And the executor is a key factor for asynchronous programming... so the answer may be "it is not the goal of the project to explain about async on C++".
